n2survey/lime/survey.py

```diff
@@ -219,9 +219,50 @@ def plot(self, question, kind: str = None, **kwargs):
         raise NotImplementedError()
 
     def get_label(self, question: str) -> str:
-        """Get label for the corresponding column or group of colums"""
-        raise NotImplementedError()
+        """Get label for the corresponding column or group of colums
+
+        Args:
+            question (str): id of question to retrieve
+
+        Returns:
+            dict: dict of label mappings
+        """
+
+        question_group = self.questions[self.questions["question_group"] == question]
+        if question_group.iloc[0]["type"] == "array":
+            # Collect labels from all subquestions if "array" type
+            label_dict = {
+                index: label
+                for index, label in zip(question_group.index, question_group["label"])
+            }
+            label_dict["question_label"] = question_group.iloc[0]["question_label"]
+        else:
+            # Collect labels from only first in the group if "single-choice" or "multiple-choice" type
+            label_dict = {question: question_group.iloc[0]["label"]}
+
+        return label_dict
```

Comment:
Note that the function typing contradicts the docstring. Typing says that the function returns `str` but the docstring says  `dict`. That, I guess, shows that we understood the function differently. The idea for this function was to return a common title for the whole question, i.e. one question -> one title. It is also supposed to work for both `question_group` and `column`. For example:

```python
# For a question group we return question_label.
# returns 'Would it be possible for you to extend your current contract/stipend for the following reasons?'
survey.get_label("B6") 

# For a single column we return the label
# returns 'More time needed to complete PhD project'
survey.get_label("B6_SQ001")
```